JSQMessagesViewController/Controllers/JSQMessagesViewController.h

```diff
@@ -24,6 +24,30 @@
 #import "JSQMessagesKeyboardController.h"
 
 /**
+ * The `JSQMessagesViewControllerScrollerDelegate` protocol enables users to respond to events caused
+ * by new messages arriving.
+ */
+@protocol JSQMessagesViewControllerScrollerDelegate <NSObject>
+
+@optional
+/**
+ *  Is called when new messages arrive, and asks the delegate whether we should
+ *  scroll to these or not.
+ *
+ *  @param indexPath The indexPath of the item to scroll to
+ *
+ *  @discussion If this method is not implemented we will default to TRUE.
+ */
+- (BOOL)shouldScrollToNewlyReceivedMessageAtIndexPath:(NSIndexPath *)indexPath;
+
+/**
+ * Is called on startup, before the view appears, and asks the delegate
+ * whether we should scroll to the bottom or not.
+ */
+- (BOOL)shouldScrollToLastMessageAtStartup;
```

Comment:
I actually had a discussion with myself regarding this.
Initially I made a property for stating whether the view was loading or not, so you can state in the viewWillAppear whether you wanted to scroll to the end or not.
I ended up doing this instead, because I thought it was more clean.
Am open to suggestions though.

> On 7. mar. 2016, at 19.46, Jesse Squires <notifications@github.com> wrote:
> 
> In JSQMessagesViewController/Controllers/JSQMessagesViewController.h <https://github.com/jessesquires/JSQMessagesViewController/pull/1486#discussion_r55221503>:
> 
> > +@optional
> > +/**
> > + *  Is called when new messages arrive, and asks the delegate whether we should
> > + *  scroll to these or not.
> > + *
> > + *  @param indexPath The indexPath of the item to scroll to
> > + *
> > + *  @discussion If this method is not implemented we will default to TRUE.
> > + */
> > +- (BOOL)shouldScrollToNewlyReceivedMessageAtIndexPath:(NSIndexPath *)indexPath;
> > +
> > +/**
> > + * Is called on startup, before the view appears, and asks the delegate
> > + * whether we should scroll to the bottom or not.
> > + */
> > +- (BOOL)shouldScrollToLastMessageAtStartup;
> hm... is this necessary? can't the previous method also handle this case?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/jessesquires/JSQMessagesViewController/pull/1486/files#r55221503>.
>